pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -416,6 +416,7 @@ impl Timeline {
             .collect_vec();
         stats.level0_deltas_count = Some(level0_deltas.len());
         // Only compact if enough layers have accumulated.
+        const L0_COMPACTION_MAX_NUM_DELTA_LAYERS: usize = 60; // 256MB * 60 = 15GB
```

Comment:
This feels high.  `CONCURRENT_BACKGROUND_TASKS` is 3/4 the thread count, so we should size the limit on L0s to compact in one go accordingly.

Maybe we should just use the `compaction_threshold`?  That way we would always be compacting exactly compaction_threshold L0s, and the typical case would be the same as the rapid-writing case.